From 015fa4c961173a3e727dc1adca4d250e92f8c120 Mon Sep 17 00:00:00 2001
Message-Id: <015fa4c961173a3e727dc1adca4d250e92f8c120.1350995715.git.minovotn@redhat.com>
In-Reply-To: <e1fce7d191263c3a20b0768faf927de7aaa2bfa1.1350995715.git.minovotn@redhat.com>
References: <e1fce7d191263c3a20b0768faf927de7aaa2bfa1.1350995715.git.minovotn@redhat.com>
From: Paolo Bonzini <pbonzini@redhat.com>
Date: Thu, 18 Oct 2012 16:22:22 +0200
Subject: [PATCH 2/3] scsi-block: remove properties that are not relevant for
 passthrough

RH-Author: Paolo Bonzini <pbonzini@redhat.com>
Message-id: <1350577343-17842-3-git-send-email-pbonzini@redhat.com>
Patchwork-id: 43368
O-Subject: [RHEL 6.4 qemu-kvm PATCH 2/3] scsi-block: remove properties that are not relevant for passthrough
Bugzilla: 831102
RH-Acked-by: Laszlo Ersek <lersek@redhat.com>
RH-Acked-by: Orit Wasserman <owasserm@redhat.com>
RH-Acked-by: Kevin Wolf <kwolf@redhat.com>

scsi-block is a passthrough device and does not allow customization
of vendor, product, removable, DPOFUA, block size or any other piece of
information.  Thus, drop DEFINE_SCSI_DISK_PROPERTIES() from the
list of qdev properties.

Signed-off-by: Paolo Bonzini <pbonzini@redhat.com>
(cherry-picked from commit 038478370d5c7ed8f66fc9523bf3f60b8800fe86)
---
 hw/scsi-disk.c | 2 +-
 1 file modificato, 1 inserzione(+). 1 rimozione(-)

Signed-off-by: Michal Novotny <minovotn@redhat.com>
---
 hw/scsi-disk.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/hw/scsi-disk.c b/hw/scsi-disk.c
index e2041c5..9c4a1d8 100644
--- a/hw/scsi-disk.c
+++ b/hw/scsi-disk.c
@@ -1977,7 +1977,7 @@ static SCSIDeviceInfo scsi_disk_info[] = {
         .destroy      = scsi_destroy,
         .alloc_req    = scsi_block_new_request,
         .qdev.props   = (Property[]) {
-            DEFINE_SCSI_DISK_PROPERTIES(),
+            DEFINE_PROP_DRIVE("drive", SCSIDiskState, qdev.conf.bs),
             DEFINE_PROP_END_OF_LIST(),
         },
 #endif
-- 
1.7.11.7

